pkg/cmd/templatefuncs.go

```diff
@@ -87,23 +88,41 @@ func (c *Config) globTemplateFunc(pattern string) []string {
 }
 
 func (c *Config) includeTemplateFunc(filename string) string {
-	var absPath chezmoi.AbsPath
-	if filepath.IsAbs(filename) {
-		var err error
-		absPath, err = chezmoi.NewAbsPathFromExtPath(filename, c.homeDirAbsPath)
-		if err != nil {
-			panic(err)
-		}
-	} else {
-		absPath = c.SourceDirAbsPath.JoinString(filename)
-	}
-	contents, err := c.fileSystem.ReadFile(absPath.String())
+	contents, err := c.readFile(filename)
 	if err != nil {
 		panic(err)
 	}
 	return string(contents)
 }
 
+func (c *Config) includeTemplateTemplateFunc(filename string, args ...any) string {
+	var data any
+	switch len(args) {
+	case 0:
+		// Do nothing.
```

Comment:
FWIW the built-in `template` function passes through the current context with `.`, although I am unsure how that is implemented

Reply:
This is possible because the builtin `template` is actually a control statement (like `if` and `range`). See https://github.com/golang/go/blob/master/src/text/template/parse/parse.go#L409-L410 for the implementation. `text/template` does not provide a method for users to add new control statements.

Reply:
Turns out I can actually pass the global context with `{{ includeTemplate $file $ -}}`